### DISCUSSION TITLE
Template Serial Battery

### DISCUSSION THREAD

**Discussion by RFDarter** [mr-manuel/venus-os_dbus-mqtt-battery#20] [Ideas]
It would be nice if we could have a template serial battery that will not directly connect to an bms but instead via USB to a esp32 or similar.

That way we could aggregate multiple batterys and implement a charge logic on the esp32 instead of the config file in the venus os.

Like we can do with your mqtt battery but without depending on a Wifi or Lan connection.

I was just having a look at your Code and it seems you called your interface `template_battery` allready.
Maybe a fitting name would be `esp_bms' or `virtual_bms' instead

**Comment by mr-manuel** [comment]
> Like we can do with your mqtt battery but without depending on a Wifi or Lan connection.

You can write yourself a driver that gets the data from USB and sends it to the internal Venus OS MQTT broker. You can also try to use Node-RED to read the data from USB and send it to the internal Venus OS MQTT broker.

> I was just having a look at your Code and it seems you called your interface template_battery allready.

I don't know where you found it. I will not change any name to `virtual` or `esp_bms`, because it's called `dbus-mqtt-battery`, which definitely describes what this driver is doing.

**Comment by RFDarter** [comment]
Sry Manuel, I got that wrong. I thought for some reason that the dbus-serial-battery was also your project.

**Reply by mr-manuel** [reply]
It is, but anyway your request does not make sense.
